neigh_linux.go

```diff
@@ -176,40 +177,47 @@ func neighHandle(neigh *Neigh, req *nl.NetlinkRequest) error {
 	return err
 }
 
-// NeighList gets a list of IP-MAC mappings in the system (ARP table).
+// NeighList returns a list of IP-MAC mappings in the system (ARP table).
 // Equivalent to: `ip neighbor show`.
 // The list can be filtered by link and ip family.
 func NeighList(linkIndex, family int) ([]Neigh, error) {
 	return pkgHandle.NeighList(linkIndex, family)
 }
 
-// NeighProxyList gets a list of neighbor proxies in the system.
+// NeighProxyList returns a list of neighbor proxies in the system.
 // Equivalent to: `ip neighbor show proxy`.
 // The list can be filtered by link and ip family.
 func NeighProxyList(linkIndex, family int) ([]Neigh, error) {
 	return pkgHandle.NeighProxyList(linkIndex, family)
 }
 
-// NeighList gets a list of IP-MAC mappings in the system (ARP table).
+// NeighListFiltered returns a list of neighbour entries filtered by link, ip family, flag and state.
+func NeighListFiltered(linkIndex, family, flags int, state uint16) ([]Neigh, error) {
```

Comment:
I think we should have a structure which describe the query filter as parameter, instead of a list of (actual) supported filter params. 

Also, at least in the comment we should specify what are the bare minimum params that need to be specified.
Or would all entries be returned if no param is specified ?

Reply:
Something like
```
func NeighListFiltered(NeighFilter filter) ([]Neigh, error) {
```
we should also investigate if such a filtered query exist for other areas like routes and see if we can come out with a generic filter structure which works for all. Or at least a base iproute2 filter which other filters will embed.

Reply:
In that case we should simply use `Ndmsg`. I added another commit.